runtime/doc/lua.txt

```diff
@@ -2957,7 +2957,16 @@ vim.fs.joinpath({...})                                     *vim.fs.joinpath()*
 vim.fs.normalize({path}, {opts})                          *vim.fs.normalize()*
     Normalize a path to a standard format. A tilde (~) character at the
     beginning of the path is expanded to the user's home directory and
-    environment variables are also expanded.
+    environment variables are also expanded. `.` and `..` components are also
+    resolved, except when the path is relative and trying to resolve it would
+    result in an absolute path. For example: the following will not get
+    resolved:
+    • `.` as the only part in a relative path (e.g., `.`, `././`).
+    • `..` in a relative path when it leads outside the current directory
+      (e.g., `../../foo`).
+
+    NOTE: In the root directory, resolving `..` will return the root directory
+    itself.
```

Comment:
- explicit input/output would be less ambiguous, less verbose, and more intuitive.
- using "" instead of backticks is more readable for this case.

```suggestion
    result in an absolute path. 
    • "." as the only part in a relative path:
        • "." => "."
        • "././" => "././"
    • ".." when it leads outside the current directory
        • "../../foo" => "../foo"
    • ".." in the root directory returns the root directory.
        • "../" => "/"
```

Reply:
That's way better, thanks a lot!